discord/automod.py

```diff
@@ -0,0 +1,468 @@
+    alert_system_message_id: Optional[:class:`int`]
+        The ID of the system message that was sent to the predefined alert channel.
+    content: :class:`str`
+        The content of the message that triggered the rule.
+        Requires the :attr:`Intents.message_content` or it will always return an empty string.
+    matched_keyword: Optional[:class:`str`]
+        The matched keyword from the triggering message.
+    matched_content: Optional[:class:`str`]
+        The matched content from the triggering message.
+        Requires the :attr:`Intents.message_content` or it will always return an empty string.
+    """
+
+    __slots__ = (
+        '_state',
+        'action',
+        'rule_id',
+        'rule_trigger_type',
+        'guild_id',
+        'user_id',
+        'channel_id',
+        'message_id',
+        'alert_system_message_id',
+        'content',
+        'matched_keyword',
+        'matched_content',
+    )
+
+    def __init__(self, *, data: AutoModerationActionExecutionPayload, state: ConnectionState) -> None:
+        self._state: ConnectionState = state
+        self.message_id: Optional[int] = utils._get_as_snowflake(data, 'message_id')
+        self.action: AutoModRuleAction = AutoModRuleAction.from_data(data['action'])
+        self.rule_id: int = int(data['rule_id'])
+        self.rule_trigger_type: AutoModRuleTriggerType = try_enum(AutoModRuleTriggerType, data['rule_trigger_type'])
+        self.guild_id: int = int(data['guild_id'])
+        self.channel_id: Optional[int] = utils._get_as_snowflake(data, 'channel_id')
+        self.user_id: int = int(data['user_id'])
+        self.alert_system_message_id: Optional[int] = utils._get_as_snowflake(data, 'alert_system_message_id')
+        self.content: str = data['content']
+        self.matched_keyword: Optional[str] = data['matched_keyword']
+        self.matched_content: Optional[str] = data['matched_content']
+
+    def __repr__(self) -> str:
+        return f'<AutoModRuleExecution rule_id={self.rule_id} action={self.action!r}>'
+
+    @property
+    def guild(self) -> Guild:
+        """:class:`Guild`: The guild this action was taken in."""
+        return self._state._get_or_create_unavailable_guild(self.guild_id)
+
+    @property
+    def channel(self) -> Optional[Union[GuildChannel, Thread]]:
+        """Optional[Union[:class:`abc.GuildChannel`, :class:`Thread`]]: The channel this action was taken in."""
+        if self.channel_id:
+            return self.guild.get_channel(self.channel_id)
+        return None
+
+    @property
+    def member(self) -> Optional[Member]:
+        """Optional[:class:`Member`]: The member this action was taken against /who triggered this rule."""
+        return self.guild.get_member(self.user_id)
+
+    async def fetch_rule(self) -> AutoModRule:
+        """|coro|
+
+        Fetch the rule whose action was taken.
+
+        You need the :attr:`Permissions.manage_guild` permission to use this.
```

Comment:
```suggestion
        You must have the :attr:`Permissions.manage_guild` permission to use this.
```

The "must have" wording is consistent (to my knowledge) across all methods in the library.